documentation/cobigen-core_configuration.asciidoc

```diff
@@ -378,6 +378,119 @@ However, if we want to reference an increment that it is not defined inside our
+     |- src
+       |- main
+         |- templates
+           |- templates
+             |- context.xml
+             |- templates.xml
+   |- template-set2
+```
+
+The updated structure splits the context.xml and the pom.xml files into multiple directories - one for each template set
+ - each with their own atomized/small/more understandable version so you can download them individually instead of all
+ at once
+
+If the configuration needs to be updated the user will be asked, whether he wants to perform an automatic upgrade. The
+automatic upgrade process can be postponed, but it is highly recommended to perform it.
+
+=== Manual Upgrade Steps
+If you wish to manually split your templates into smaller parts, as specified in version 3.0, please follow the following instruction.
+Navigate to your CobiGen directory e.g. `$cghome/.cobigen`. If the `template-sets` folder is missing,
+please create one. Navigate into the folder and create a `downloaded` and an `adapted` folder.
+
+1. Navigate to the newly created `adapted` folder and create a new folder named after the template you want to upgrade
+e. g. `template-set1`
+
+2. In the newly created folder, create the following folder-structure:
+```
+ |- adapted
+   |- template-set1
+     |- src
+       |- main
+         |- templates
+```
+Whereas `template-set1` is the name of the template you want to upgrade.
+[start=3]
+3. Within the old templates directory, navigate to `CobiGen_Templates\src\main\java` and copy the `java` folder into your newly created `main` folder.
+The structure of the `adapted` folder should now look like this:
+```
+ |- adapted
+   |- template-set1
+     |- src
+       |- main
+         |- java
+         |- templates
+```
+[start=4]
+4. Navigate to the template you want to upgrade e.g. `CobiGen_Templates\src\main\templates\template-set1` and copy all of it's contents
+into your newly generated templates folder (`template-sets\adapted\templateSet1\src\main\templates`).
+
+6. Switch to `template-sets\adapted\templateSet1\src\main\templates` and create a new 'context.xml' file, with the following content:
+```
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<contextConfiguration xmlns="http://capgemini.com/devonfw/cobigen/ContextConfiguration" version="3.0">
+    <trigger id="templateSet1" type="java" templateFolder="templateSet1" inputCharset="UTF-8">
+        <containerMatcher type="package" value="someValue" retrieveObjectsRecursively="false"/>
+        <matcher type="fqn" value="*" accumulationType="NOT">
+            <variableAssignment type="constant" key="foo" value="BLUBB1"/>
+        </matcher>
+    </trigger>
+    <tags>
+        <tag name="PLACEHOLDER---This tag was inserted through the upgrade process and has to be changed manually---PLACEHOLDER"/>
+    </tags>
+    <links>
+        <link url="PLACEHOLDER---This tag was inserted through the upgrade process and has to be changed manually---PLACEHOLDER"/>
+    </links>
+</contextConfiguration>
```

Comment:
Should be replaced with new templateSetConfiguration introduced in #1495.

Reply:
Where can I find the new templateSetConfiguration?